Signal/src/view controllers/ContactsPicker.swift

```diff
@@ -307,25 +294,29 @@ open class ContactsPicker: UITableViewController, UISearchResultsUpdating, UISea
     }
     
     // MARK: - Search Actions
+
+    open func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
+        searchBar.resignFirstResponder();
+    }
+
+    open func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+        updateSearchResults(searchText: searchText)
+    }
     
-    open func updateSearchResults(for searchController: UISearchController) {
-        if let searchText = resultSearchController.searchBar.text , searchController.isActive {
-            
-            let predicate: NSPredicate
-            if searchText.characters.count == 0 {
-                filteredSections = sections
-            } else {
-                do {
-                    predicate = CNContact.predicateForContacts(matchingName: searchText)
-                    let filteredContacts = try contactStore.unifiedContacts(matching: predicate, keysToFetch: allowedContactKeys)
+    open func updateSearchResults(searchText: String) {
+        let predicate: NSPredicate
+        if searchText.characters.count == 0 {
+            filteredSections = sections
+        } else {
+            do {
+                predicate = CNContact.predicateForContacts(matchingName: searchText)
+                let filteredContacts = try contactStore.unifiedContacts(matching: predicate,keysToFetch: allowedContactKeys)
                     filteredSections = collatedContacts(filteredContacts)
-                } catch let error as NSError {
-                    Logger.error("\(self.TAG) updating search results failed with error: \(error)")
-                }
+            } catch let error as NSError {
+                Logger.error("\(self.TAG) updating search results failed with error: \(error)")
             }
-
-            self.tableView.reloadData()
         }
+        self.tableView.reloadData()
     }
     
     open func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
```

Comment:
I no longer see a way to cancel the search.

Previously:
![search-filter](https://cloud.githubusercontent.com/assets/217057/21268622/286aead4-c375-11e6-81fc-982a1f0df2fd.jpg)